alf/algorithms/icm_algorithm.py

```diff
@@ -134,9 +163,15 @@ def train_step(self, inputs, state, calc_intrinsic_reward=True):
                 intrinsic_reward)
 
         return AlgorithmStep(
-            outputs=intrinsic_reward,
+            outputs=(),
             state=feature,
-            info=LossInfo(
-                loss=forward_loss + inverse_loss,
-                extra=dict(
-                    forward_loss=forward_loss, inverse_loss=inverse_loss)))
+            info=ICMInfo(
+                reward=intrinsic_reward,
+                loss=LossInfo(
+                    loss=forward_loss + inverse_loss,
+                    extra=dict(
+                        forward_loss=forward_loss,
+                        inverse_loss=inverse_loss))))
+
+    def calc_loss(self, info: ICMInfo):
```

Comment:
should avoid masking out across-episode transitions by converting the loss to scalar. Maybe define a wrapper called mask_invariant_loss()

Reply:
added reduce_mean